global/scripts/TopupPreprocessingAll.sh

```diff
@@ -120,61 +161,205 @@ fi
 ${FSLDIR}/bin/imcp $PhaseEncodeOne ${WD}/PhaseOne
 ${FSLDIR}/bin/imcp $PhaseEncodeTwo ${WD}/PhaseTwo
 ${FSLDIR}/bin/imcp $ScoutInputName ${WD}/SBRef
+log_Msg "SBRef: $ScoutInputName"
+log_Msg "PhaseOne: $PhaseEncodeOne"
+log_Msg "PhaseTwo: $PhaseEncodeTwo"
+
+if [[ ! -z $PhaseEncodeOne2 && $PhaseEncodeOne2 != NONE ]] ; then 
+    if [[ `fslhd $PhaseEncodeOne2 | grep '^dim[123]'` != `fslhd $ScoutInputName | grep '^dim[123]'` ]]
+    then
+        log_Err_Abort "2nd pair of spin echo fieldmap images have different dimensions!"
+    fi
+    ${FSLDIR}/bin/imcp $PhaseEncodeOne2 ${WD}/PhaseOne2
+    ${FSLDIR}/bin/imcp $PhaseEncodeTwo2 ${WD}/PhaseTwo2
+    log_Msg "PhaseOne2: $PhaseEncodeOne2"
+    log_Msg "PhaseTwo2: $PhaseEncodeTwo2"
+    Phase2ndDir=TRUE
+else
+    Phase2ndDir=FALSE
+fi
```

Comment:
The `log_Msg` lines added here are redundant with `opts_ShowValues`, this is why none of the other parameters get logged manually.

```suggestion

if [[ ! -z $PhaseEncodeOne2 && $PhaseEncodeOne2 != NONE ]] ; then 
    if [[ `fslhd $PhaseEncodeOne2 | grep '^dim[123]'` != `fslhd $ScoutInputName | grep '^dim[123]'` ]]
    then
        log_Err_Abort "2nd pair of spin echo fieldmap images have different dimensions!"
    fi
    ${FSLDIR}/bin/imcp $PhaseEncodeOne2 ${WD}/PhaseOne2
    ${FSLDIR}/bin/imcp $PhaseEncodeTwo2 ${WD}/PhaseTwo2
    Phase2ndDir=TRUE
else
    Phase2ndDir=FALSE
fi
```

Reply:
Thank you for your suggestion. I've fixed it.
374f6b1b2192d35b516a84898778587745f988d8